### ISSUE TITLE
ttl values - signed or unsigned

### ISSUE THREAD

**Issue by kanurag94** [c-ares/c-ares#547]
According to [RFC 2181](https://datatracker.ietf.org/doc/html/rfc2181#section-8) -- TTL values are unsigned integer with a minimum value of 0 and a max of 2^31 - 1. 
The [current c-ares library mentions the field of type `int`](https://github.com/c-ares/c-ares/blob/main/include/ares.h#L613) -- I think we should make it unsigned int.

I am not aware of the c-ares implementation in detail and don't know the implication of this change. But I would love to volunteer to take this up if maintainers feel this is a valid issue to be worked upon.

**Comment by kanurag94** [comment]
I guess that makes sense. I recently bumped into an issue where I was trying to write a function to limit the max_ttl to 2^31-1, and assigned `ttl = INT_MAX + 1` which naturally overflowed. Then I quickly verified my expectations with the RFC.
Maybe we leave a comment there or not since its really clean.

I agree this shouldn't really be an issues but would have been a proper addition if done at get-go. Closing the issue.
